### PR TITLE
[bitcore-lib] add some missing types for Address

### PR DESCRIPTION
**Pull request by dr-erych**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/bitpay/bitcore/blob/master/packages/bitcore-lib/lib/address.js
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header. (only brings them up to date partially)

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## bitcore-lib ([<kbd>unpkg</kbd>](https://unpkg.com/browse/bitcore-lib@latest/))
was missing the following properties:
1. version
1. versionGuard
1. encoding
1. util
1. errors

<details>
<summary>as well as these 5 other properties...</summary>
<p>MerkleBlock, BlockHeader, Opcode, URI, deps</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c87eacec15426c374b2ddf4d4949be4be812ce32
</p>

**Comment by MysteryBlokHed** [comment]
I think these values are missing from the types:

<https://github.com/bitpay/bitcore/blob/bed0702ff56ca207a09711fd2e2a2b75374b51ac/packages/bitcore-lib/lib/address.js#L126-L135>

File: types/bitcore-lib/index.d.ts [hunk 1]
```diff
@@ -292,7 +292,44 @@ export class Address {
...
+    static fromScriptHash(hash: Buffer, network: string | Networks.Network): Address;
+    static fromScript(script: Script, network: string | Networks.Network): Address;
+    static fromBuffer(buffer: Buffer, network?: string | Networks.Network, type?: string): Address;
+    static fromString(str: string, network?: string | Networks.Network, type?: string): Address;
+    static fromObject(obj: object): Address;
+
+    static getValidationError(data: string, network?: string | Networks.Network, type?: string): null | Error;
+
+    toBuffer(): Buffer;
+    toObject(): object;
```
Comment (MysteryBlokHed) [inline types/bitcore-lib/index.d.ts:322]: This return type can probably be more specific: <https://github.com/bitpay/bitcore/blob/bed0702ff56ca207a09711fd2e2a2b75374b51ac/packages/bitcore-lib/lib/address.js#L615-L619>

```suggestion
    toObject(): {
        hash: string;
        type: string;
        network: string;
    };
```

File: types/bitcore-lib/index.d.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for bitcore-lib 0.15
+// Type definitions for bitcore-lib 8.25
```
Comment (MysteryBlokHed) [inline types/bitcore-lib/index.d.ts:1]: There are probably a lot of API changes from v0.15 to v8.25 that won't be included in this PR

**Review by sandersn — CHANGES_REQUESTED** [review]
Requesting changes so the bot will take @MysteryBlokHed 's review seriously.

**Pull request closed without merging**